### ISSUE TITLE
[Feature request] Display file contents of sub directories in project directory

### ISSUE THREAD

**Issue by Brakjen** [texstudio-org/texstudio#639]
### Displaying directory contents in the Structure side bar in TeXstudio.

I personally, and I suspect many other users as well, structure our projects with directories dedicated to various file types. For example, I always have a "./figs" dir where I store my figure files. Having the possibility of displaying the file contents of such dirs directly from within TeXstudio, would make my life a lot simpler. Now I have to keep open a Finder window or terminal window to display the file names, instead of just looking in the side bar.

Having TeXstudio also autocomplete the filenames of files in pre-defined, standard directories, would also be an awesome feature. The dir "./figs" could be one such default place where to look for files. Maybe we could set a few of these in the settings?

**Comment by sunderme** [comment]
I don't fully understand. txs completes filenames, hence when you type './fi' , complete it to 'figs', you can complete further filenames. What should txs do instead ?

**Comment by muzimuzhi** [comment]
@Andersmb 
> Having the possibility of displaying the file contents of such dirs directly from within TeXstudio, would make my life a lot simpler.

I think texstudio has already supported this feature. Here is a step-by-step guideline
 1. suppose you are using `\includegraphics{<cursor position>}`
 1. keep cursor position inside curly braces
 1. hit shortcut of (on macOS it's `Ctrl + space`) or click menu item `Idefix -> Complete -> Normal`, then the completer pops up
 1. choose the sub directory containing figures, hit "enter"
 1. continue to choose the desired figure you want to include.

![Screen Recording 2019-07-06 at 01 45 31](https://user-images.githubusercontent.com/6376638/60738614-8a3f7680-9f91-11e9-8939-1a092d31d587.gif)

**Comment by ludwigbald** [comment]
It would be nice to see the filesystem structure in the tree on the left, where we currently only see the structure of the currently loaded LaTeX documents. There's plenty of space left to display the contents of the whole workspace folder.

**Comment by ankyAgrawal** [comment]
Need the ability to see the project file structure in the left panel.

**Comment by YunfeiZHAO** [comment]
It would be really nice to see the filesystem structure in the tree on the left like on overleaf!

**Comment by dlabaja** [comment]
I honestly feel like every LaTeX editor should have this feature

**Comment by athn-nik** [comment]
Actually this would be super useful and needed. The current view is far from intuitive

**Comment by sunderme** [comment]
first version implemented (https://github.com/texstudio-org/texstudio/releases/tag/4.8.0alpha3)

**Comment by athn-nik** [comment]
Thanks for taking care of this, but how to test this feature? How to turn on the normal file structure in case I install the prereleased version @sunderme ?

**Comment by sunderme** [comment]
You have tabs on the left-hand side to select which functionaliyt to use. Furthermore the docks can be split.

**Comment by dlabaja** [comment]
thanks for this work, texstudio will be finally a really good alternative to overleaf

**Comment by pavelkosov99** [comment]
How to show filesystem on the left as in overleaf? It was shown but I closed it and now I cannot get it back

**Comment by sunderme** [comment]
Can you post a screenshot?

04.04.2024 02:24:04 Pavel Kosov ***@***.***>:

**Comment by octaeder** [comment]
Maybe this is a misunderstanding. This is not a general file browser. When a tex file is opened with txs then the location of the file is used as top node for the file viewer. So you can navigate to everything that is in that subtree, but not to things outside - especially not the parent of the subtree. Said this, be aware that closing the editor will remove the subtree.

**Comment by yagoiroot** [comment]
It's unclear to me from this thread what the current state of this feature request is. Currently, as far as I can tell there is no way to view the files in the project directory from TeXstudio. 

This is a feature that exists in many IDEs as well as other dedicated LaTeX editor applications, such as Overleaf. The advantage of this is that you can open other files in the project without having to navigate to any other windows, allowing for switching between editing different files. TeXstudio's feature to have different files open in different tabs goes part way to addressing this issue (other editor only support having one file "open" at a time). However, it requires going to another window to open additional files in the first place. 

If there was a way to view the files in the project directory, this would be very nice benefit for ease of usability.

**Comment by octaeder** [comment]
When you open a project then the file side view will show the folder of the root document. You may need to select this view with the context menu when it is not present in the icons list.

**Comment by yagoiroot** [comment]
@octaeder Thank you for the prompt reply. I attempted to locate the feature, as you described, on the two different machines I use which have slightly different TeXstudio instantiations. I found it on both and on one it does indeed work, to show the files in the project directory, while on the other it did not display the files (as if the directory was empty). This is mostly likely some bug on my end, but, regardless, thank you for confirming that the feature is indeed present.
